docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
+
+## Current Challenge
+The lack of clear criteria for classifying components as core or non-core has led to:
+- Inconsistent decision-making across different features and plugins
+- Potential bloat in core components
+- Uncertainty in plugin development and maintenance strategies
+
+## Proposed Criteria for Core Components
+
+A component should be considered "core" if it meets the following criteria:
+
+### 1. Essential Functionality
+- Necessary for fundamental platform capabilities, aka required by core plugins / more than 5 external plugins, like saved objects management, embeddable, data plugin.
```

Comment:
> required by core plugins / more than 5 external plugins

I think 5 external plugins might be hard to measure, and not every core plugin will be required by a core plugin. Plugins like devtools probably won't have any downstream dependency but it should be in core

Reply:
Yeah, agree that. The bar for 5 plugins is too high, we can make it 2 maybe.